GT-124 Make references to operational tools configurable

rowan04 opened the pull request:
- added config options for helpdesk link, helpdesk support unit, request tracker and community documentation
- so that different instances of gocdb can have links point directly to their tooling, instead of having general references

rowan04 committed:
Make references to operational tools configurable

- added config options for helpdesk link, helpdesk support unit, request tracker and community documentation
- so that different instances of gocdb can have links point directly to their tooling, instead of having general references

rowan04 committed:
Edits to pass code tests

- shortened variable name
- fixed href formatting
- still doen't work as planned - needs a php file rework
- this commit was made as I'm about to leave for christmas

rowan04 committed:
Changes Help_And_Contact page from .html to .php

- in order to have configurable links in Help_And_Contact,
- the file had to be changed from .html to .php
- so it was also moved to the static_php directory
- the operationalLinks file was made to call the config file and assign to variables
- this file is then called in Help_And_Contact

rowan04 committed:
Allows Help_And_Contact.php to be displayed

- as Help_And_Contact page was moved from a .html file to .php,
- the methods that display it on the gocdb web needed to be changed
- as it is no longer a static_html file

rowan04 commented:
Note - this currently doesn't work and the gocdb webpage doesn't have the correct links. I am still working on this

rowan04 commented on htdocs/web_portal/index.php:
```diff
@@ -208,6 +208,11 @@ function Draw_Page($Page_Type) {
             break;
+        case "Static_PHP":
+            //rejectIfNotAuthenticated();
+	    //require_once __DIR__.'/components/Draw_Components/draw_page_components.php';
```
This line is to be removed

rowan04 commented on htdocs/web_portal/index.php:
```diff
@@ -686,6 +712,25 @@ function Get_Static_Page_Contents($Page_Name) {
+    if(!isset($Available_Static_Pages[$PHP_Page_Name])) {
+        return "";
+    }
+    $PHP = Get_File_Contents($phpDir."/".$PHP_Page_Name);
```
Here's the problem (line 724) - Get_File_Contents just gets the contents of the file, it doesn't execute it. Therefore the php parts aren't being executed, just written out. idk what to replace it with though

rowan04 replied:
Fixed this - I just used the commented out statements below I'd tried before! I had to use the `systemctl reload httpd` to update it, which is maybe why it hadn't worked before.

rowan04 committed:
Fixes configurable links not working on webpage

- changed method of ouputting the Help_And_Contact.php file
- so it now outputs the executed file, not the file contents
- meaning the php parts within the html are properly executed, and not just output

rowan04 committed:
Tidies up code and improves comments

- removes unnecessary commented out code
- adds to existing comments
- fixes out of date code in search_ses.php

gregcorbett commented:
Hmmmmm, I wonder if moving the Help_And_Contact to be fully within the view/controller setup would be a cleaner solution than using `ob_start()` et al.

my_sites.php is a relatively small example of the view/controller setup in use:
- https://github.com/GOCDB/gocdb/blob/dev/htdocs/web_portal/controllers/my_sites.php
- https://github.com/GOCDB/gocdb/blob/dev/htdocs/web_portal/views/my_sites.php

So for this, you'd have to create, for example:
- `htdocs/web_portal/controllers/help_and_contact.php`. 
- `htdocs/web_portal/views/help_and_contact.php`. 

The controller is where the code in htdocs/web_portal/static_php/operationalLinks.php would go, except with the output contained in a `params` array.

The view is where the contents of `htdocs/web_portal/static_php/Help_And_Contact.php` would go, except with references to the `params` array, i.e. `params['HelpdeskLink']` instead of `$helpdeskLink`.

You'd have to add an additonal case for PageType=Help_And_Contact in [web_portal/index.php](https://github.com/GOCDB/gocdb/blob/dev/htdocs/), similar to the [My_Sites case](https://github.com/GOCDB/gocdb/blob/dev/htdocs/web_portal/index.php#L300), but it would mean less new code in index.php.

Could you give that a go?

rowan04 commented:
> Hmmmmm, I wonder if moving the Help_And_Contact to be fully within the view/controller setup would be a cleaner solution than using `ob_start()` et al.
> 
> my_sites.php is a relatively small example of the view/controller setup in use:
> 
> * https://github.com/GOCDB/gocdb/blob/dev/htdocs/web_portal/controllers/my_sites.php
> * https://github.com/GOCDB/gocdb/blob/dev/htdocs/web_portal/views/my_sites.php
> 
> So for this, you'd have to create, for example:
> 
> * `htdocs/web_portal/controllers/help_and_contact.php`.
> * `htdocs/web_portal/views/help_and_contact.php`.
> 
> The controller is where the code in htdocs/web_portal/static_php/operationalLinks.php would go, except with the output contained in a `params` array.
> 
> The view is where the contents of `htdocs/web_portal/static_php/Help_And_Contact.php` would go, except with references to the `params` array, i.e. `params['HelpdeskLink']` instead of `$helpdeskLink`.
> 
> You'd have to add an additonal case for PageType=Help_And_Contact in [web_portal/index.php](https://github.com/GOCDB/gocdb/blob/dev/htdocs/), similar to the [My_Sites case](https://github.com/GOCDB/gocdb/blob/dev/htdocs/web_portal/index.php#L300), but it would mean less new code in index.php.
> 
> Could you give that a go?

ok

gregcorbett reviewed (changes requested):
see my previous comment

rowan04 committed:
Change Help_And_Contact.php page to use a view/controller setup instead of a static_php setup

- moved static_php/Help_And_Contact.php to views/help_and_contact.php
- removed static_php/operationalLinks and replaced it with controllers/help_and_contact.php
- this was done so that there wasn't lots of new code for the static_php method in index.php

rowan04 committed:
Changes to files to work with the new view/controller setup for help_and_contact.php

- change to menu.xml so that it uses correct index.php case
- change index.php case to use the new view/controller setup
- change search_ses.php to use the correct new link

rowan04 committed:
Change to pass code tests

- adds missed \ in search_ses.php

gregcorbett reviewed (approved):
This looks great :). I want to prepare the necassary config change, in a sandbox, at the same time I merge this.

gregcorbett committed:
Fix (some of the) lint identified by codacy
